examples/mfa/mfacosigner/mfacosigner.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 OpenPubkey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package mfacosigner
+
+import (
+	"crypto"
+	"crypto/rand"
+	"fmt"
+
+	"github.com/go-webauthn/webauthn/protocol"
+	"github.com/go-webauthn/webauthn/webauthn"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/openpubkey/openpubkey/cosigner"
+)
+
+type UserKey struct {
+	Issuer string // ID Token issuer (iss)
+	Aud    string // ID Token audience (aud)
+	Sub    string // ID Token subject ID (sub)
+}
+
+func NewUser(as *cosigner.AuthState) *user {
+	return &user{
+		id:          []byte(as.Sub),
+		username:    as.Username,
+		displayName: as.DisplayName,
+	}
+}
+
+// This is intended as an example. Both sessionMap and users are not concurrency safe.
+type MfaCosigner struct {
+	*cosigner.AuthCosigner
+	webAuthn   *webauthn.WebAuthn
+	sessionMap map[string]*webauthn.SessionData
+	users      map[cosigner.UserKey]*user
+}
+
+func New(signer crypto.Signer, alg jwa.SignatureAlgorithm, issuer, keyID string, cfg *webauthn.Config) (*MfaCosigner, error) {
+	hmacKey := make([]byte, 64)
+
+	if _, err := rand.Read(hmacKey); err != nil {
+		return nil, err
+	}
+
+	wauth, err := webauthn.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	authCos, err := cosigner.New(signer, alg, issuer, keyID)
```

Comment:
Per my comment elsewhere, this is where we should create inject the `AuthStateInMemoryStore`

Reply:
Should be fixed now